src/compositor/compositor-xrender.c

```diff
@@ -584,6 +584,8 @@ make_shadow (MetaDisplay   *display,
         }
     }
 
+  g_free (data);
```

Comment:
You can't do `g_free (data);` here, **ximage->data** points to **data**. 
https://tronche.com/gui/x/xlib/utilities/XCreateImage.html

Reply:
ok, this new patch works:

```patch
diff --git a/src/compositor/compositor-xrender.c b/src/compositor/compositor-xrender.c
index c18c8dd..de112db 100644
--- a/src/compositor/compositor-xrender.c
+++ b/src/compositor/compositor-xrender.c
@@ -584,6 +584,8 @@ make_shadow (MetaDisplay   *display,
         }
     }
 
+  ximage->data = (char *) data;
+
   return ximage;
 }
```

@rbuj  are you agree?

Reply:
yes

Reply:
@sc0w what `ximage->data = (char *) data;` is supposed to do?

Reply:
@rbuj ok, done!

@muktupavels it fixes the warning I mentioned

Reply:
Then perhaps add comment explaining need for that? Because otherwise it is useless code... It does nothing, no? https://gitlab.freedesktop.org/xorg/lib/libx11/blob/master/src/ImUtil.c#L360